### DISCUSSION TITLE
Repeater as a relationship not working

### DISCUSSION THREAD

**Discussion by ArtMin96** [filamentphp/filament#9270] [Help]
Could you tell me what I did wrong here? I did everything as it was done in the filament demo, but for some reason, the repeater understands it as a column of `orders`.
There is no `services` column in the `orders` table.
I'm sorry that I didn't send the error with the flare. There are things there that I don't want to make public.

![QkQtw1K](https://github.com/filamentphp/filament/assets/29732308/b14e6953-d507-4477-985d-c9d3afa4bf2c)

Order model
```php
public function services(): HasMany
{
    return $this->hasMany(OrderService::class, 'order_id');
}
```

```php
class OrderService extends Model
{
    protected $fillable = [
        'service_id',
        'order_id',
        // ...
    ];
}
```

OrderResource
```php
Repeater::make('services')
    ->relationship()
    ->schema([
        // ...
    ]),
```

**Comment by VincvanAs** [comment]
Hi

Try using

`->relationship('services')`

On your repeater

```
Repeater::make('service_select')
    ->relationship('services')
    ->schema([
        // ...
    ]),
```

**Reply by ArtMin96** [reply]
Thanks @VincvanAs I tried that too.
**SQLSTATE[42S22]: Column not found: 1054 Unknown column 'service_select' in 'field list'**

**Reply by ArtMin96** [reply]
It started working. I don't know what was causing the problem, but I changed the name of the relation and it worked even though there was nothing else called `services` in the `Order` model.

```php
Repeater::make('services')
    ->relationship('serviceItems')
```

**Reply by leandrocfe** [reply]
What about
```php
Repeater::make('serviceItems') ->relationship()
```
?

**Reply by ArtMin96** [reply]
@leandrocfe not working :confused:

**Reply by leandrocfe** [reply]
Are using resources or custom page?

**Reply by ArtMin96** [reply]
Resources
